pymc_extras/statespace/models/DFM.py

```diff
@@ -0,0 +1,828 @@
+        error_order: int = 0,
+        error_var: bool = False,
+        error_cov_type: str = "diagonal",
+        measurement_error: bool = False,
+        verbose: bool = True,
+    ):
+        if k_endog is None and endog_names is None:
+            raise ValueError("Either k_endog or endog_names must be provided.")
+        if k_endog is None:
+            k_endog = len(endog_names)
+        if endog_names is None:
+            endog_names = [f"endog_{i}" for i in range(k_endog)]
+
+        self.endog_names = endog_names
+        self.k_endog = k_endog
+        self.k_factors = k_factors
+        self.factor_order = factor_order
+        self.error_order = error_order
+        self.error_var = error_var
+        self.error_cov_type = error_cov_type
+
+        if k_exog is None and exog_names is None:
+            self._exog = False  # flag for presence of exogeneous variables
+            self.k_exog = 0
+        else:
+            self._exog = True
+            self.shared_exog_states = shared_exog_states
+            self.exog_innovations = (
+                exog_innovations if exog_innovations is not None else False
+            )  # default if not provided is False
+            if k_exog is None:
+                k_exog = len(exog_names) if exog_names is not None else 0
+            elif exog_names is None:
+                exog_names = [f"exog_{i}" for i in range(k_exog)] if k_exog > 0 else None
+            self.k_exog_states = k_exog * k_endog if not shared_exog_states else k_exog
+
+        self.exog_names = exog_names
+        self.k_exog = k_exog
+
+        # TODO add exogenous variables support
+        # I start implementing a version of exog support based on pymc_extras/statespace/models/structural/components/regression.py
+        # exog_innovations control if the beta coefficient follows a random walk
+        # shared_exog_states control if the exogenous states are shared across equations
+        # I tested the case of shared_exog_states=True and exog_innovations=False vs the stats case by looking at trajectory and everything works well
+
+        # Determine the dimension for the latent factor states.
+        # For static factors, one use k_factors.
+        # For dynamic factors with lags, the state include current factors and past lags.
+        # If factor_order is 0, we treat the factor as static (no dynamics),
+        # but it is still included in the state vector with one state per factor.
+        # Factor_ar paramter will not exist in this case.
+        k_factor_states = max(self.factor_order, 1) * k_factors
+
+        # Determine the dimension for the error component.
+        # If error_order > 0 then we add additional states for error dynamics, otherwise white noise error.
+        k_error_states = k_endog * error_order if error_order > 0 else 0
+
+        # Total state dimension
+        k_states = k_factor_states + k_error_states + (self.k_exog_states if self._exog else 0)
```

Comment:
```suggestion
        k_states = k_factor_states + k_error_states + self.k_exog_states
```

`k_exog_states` is already zero if self._exog is False right?